lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -2134,24 +2134,32 @@ FModuleOp buildInnerFIFO(CircuitOp circuit, StringRef moduleName,
   };
 
   // Signal declarations
-  auto zeroConst =
-      createConstantOp(bitType, APInt(1, 0), builder.getLoc(), builder);
+  auto zeroConstPtr = createConstantOp(
+      depthPtrType, APInt(depthPtrType.getBitWidthOrSentinel(), 0),
+      builder.getLoc(), builder);
+  auto zeroConstDepth =
+      createConstantOp(depthType, APInt(depthType.getBitWidthOrSentinel(), 0),
+                       builder.getLoc(), builder);
```

Comment:
The handshake dialect should probably at some point switch over to lowering to HW instead of FIRRTL :thinking: